pkg/exporter/process.go

```diff
@@ -42,35 +42,71 @@ type ExportingProcess struct {
 	connToCollector net.Conn
 	obsDomainID     uint32
 	seqNumber       uint32
+	templateID      uint16
 	set             *entities.Set
 	msg             *entities.MsgBuffer
 	templatesMap    map[uint16]templateValue
+	templateRefCh   chan struct{}
 }
 
-func InitExportingProcess(collectorAddr net.Addr, obsID uint32) (*ExportingProcess, error) {
+// InitExportingProcess takes in collector address(net.Addr format), obsID(observation ID) and tempRefTimeout
+// (template refresh timeout). tempRefTimeout is applicable only for collectors listening over UDP. For TCP, you can
+// pass any value. For UDP, if 0 is passed, consider 1800s as default.
+// TODO: Get obsID, tempRefTimeout as args which can be of dynamic size supporting both TCP and UDP.
+func InitExportingProcess(collectorAddr net.Addr, obsID uint32, tempRefTimeout uint32) (*ExportingProcess, error) {
 	conn, err := net.Dial(collectorAddr.Network(), collectorAddr.String())
 	if err != nil {
 		log.Printf("Cannot the create the connection to configured ExportingProcess %s. Error is %v", collectorAddr.String(), err)
 		return nil, err
 	}
 	msgBuffer := entities.NewMsgBuffer()
-	return &ExportingProcess{
+
+	expProc := &ExportingProcess{
 		connToCollector: conn,
 		obsDomainID:     obsID,
 		seqNumber:       0,
+		templateID:      startTemplateID,
 		set:             entities.NewSet(msgBuffer.GetMsgBuffer()),
 		msg:             msgBuffer,
 		templatesMap:    make(map[uint16]templateValue),
-	}, nil
+		templateRefCh:   make(chan struct{}),
+	}
+
+	// Template refresh logic is only for UDP transport.
+	if collectorAddr.Network() == "udp" {
+		if tempRefTimeout == 0 {
+			// Default value is 1800s
+			tempRefTimeout = 1800
+		}
+		go func() {
+			ticker := time.NewTicker(time.Duration(tempRefTimeout) * time.Second)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-expProc.templateRefCh:
+					break
+				case <-ticker.C:
+					err := expProc.sendRefreshedTemplates()
+					if err != nil {
+						// Other option is sending messages through channel to library consumers
+						log.Printf("Error when sending refreshed templates. Closing the connection to IPFIX controller")
```

Comment:
Will it be better if we also print the error message in the log?

Reply:
There is no error message in log library. Only options are fatal, panic and print helpers.

Reply:
I mean something like `log.Printf("Error xxx %v", err)` so that we can have a better idea of what the err is.